incubator/hnc/pkg/validators/object_test.go

```diff
@@ -0,0 +1,64 @@
+package validators
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	api "github.com/kubernetes-sigs/multi-tenancy/incubator/hnc/api/v1alpha1"
+	"github.com/kubernetes-sigs/multi-tenancy/incubator/hnc/pkg/forest"
+)
+
+func TestObject(t *testing.T) {
+	f := forest.NewForest()
+	o := &Object{Forest: f}
+	l := zap.Logger(false)
+
+	tests := []struct {
+		name     string
+		oldLabel string
+		oldValue string
+		newLabel string
+		newValue string
+		fail     bool
+	}{
+		{name: "LabelInheritedFrom label doesn't exist",
+			oldLabel: "oldLabel", oldValue: "foo",
+			newLabel: "newLabel", newValue: "bar"},
+		{name: "LabelInheritedFrom label stays the same",
+			oldLabel: api.LabelInheritedFrom, oldValue: "foo",
+			newLabel: api.LabelInheritedFrom, newValue: "foo"},
+		{name: "LabelInheritedFrom label's value is changed",
```

Comment:
Nit: this is entirely up to you but I sometimes like this syntax instead (less but more consistent indentation):

```
}{{
   name: "...",
   oldLabel: "...", oldValue: "...",
   ...
}, {
   ... next testcase ...
}}
```

Entirely up to you though.

Reply:
+1

Reply:
Done.